src/store.cc

```diff
@@ -68,6 +81,9 @@ void MetadataStore::UpdateMetadata(const MonitoredResource& resource,
   // be a huge deal.
   metadata_map_.erase(resource);
   metadata_map_.emplace(resource, std::move(entry));
+
+  std::lock_guard<std::mutex> last_collection_lock(last_collection_mu_);
+  last_collection_map_.emplace(resource.type(), entry.collected_at);
```

Comment:
There may be many resources of a given type, and we want the health check to be as unobtrusive as possible. Iterating over all of the resources to extract this information may be prohibitive, especially because we're locking the map while we're iterating. Yes, this map is a cache, but I think it's justified.

Reply:
Acknowledged.